inst/rmarkdown/templates/oup/resources/template.tex

```diff
@@ -96,48 +122,76 @@
   \fi
 }
 
-\newcommand*\thanksbycode[1]{%
-  \ifcounter{FNCT@#1}
-    {\samethanks[\value{FNCT@#1}]}
-    {\thanks{\Ref{FN}{#1}}\newcounter{FNCT@#1}\setcounter{FNCT@#1}{\value{footnote}}}
+\newcommand*\addrlblbycode[1]{%
+  \ifcounter{ADDRLBL@#1}
+    {}
+    {\refstepcounter{addrcnt}\newcounter{ADDRLBL@#1}\setcounter{ADDRLBL@#1}{\value{addrcnt}}}%
+    \arabic{ADDRLBL@#1}%
+}
+
+\newcommand*\addrbycode[1]{%
+  \ifcounter{ADDR@#1}
+    {}
+    {\newcounter{ADDR@#1}%
+     \address[\addrlblbycode{#1}]{\MyRef{ADDRTXT}{#1}}}%
+}
+
+\newcommand*\corresplblbycode[1]{%
+  \ifcounter{CORRESPLBL@#1}
+    {}
+    {\refstepcounter{correspcnt}\newcounter{CORRESPLBL@#1}\setcounter{CORRESPLBL@#1}{\value{correspcnt}}}%
+    \fnsymbol{CORRESPLBL@#1}%
+}
+
+\newcommand*\correspbycode[1]{%
+  \ifcounter{CORRESP@#1}
+    {}
+    {\newcounter{CORRESP@#1}%
+     \corresp[\corresplblbycode{#1}]{\MyRef{CORRESPTXT}{#1}}}%
```

Comment:
About all the above here: Is this custom for backward compatibility ? Is it from template ? 

Trying to understand to facilitate maintainance of this in the future

Reply:
This is my code to make their template format work with author affiliations and footnotes that are specified by ID code in the Rmd skeleton. This was horrifyingly difficult to figure out, but I can't see a better way and now the code is sufficiently generic that I think it might be useful elsewhere in other rticles formats.

Reply:
We could then benefit to have that in an external file and make it available in different templates ? Is that what you are saying ?

Reply:
Could be. We would have to look to see how other formats and style files deal with multiple author affiliations and author notes. In my case, I had to build this as it isn't in the OUP CLS file itself, but I know that this sort of stuff is built into the Elsevier CLS files, avoiding the need for this sort of gymnastics.